lib/balena.js

```diff
@@ -122,6 +122,12 @@ const getSdk = function (opts) {
 		}
 	}
 
+	if ('apiKey' in opts) {
```

Comment:
Can you remind me why we still need to support this?

Reply:
Our dependencies still support using apiKeys since they are more generic than the SDK.
Since the SDK just passes the options to balena-request & balena-pine if we don't explicitly remove it, it will still reach downstream and keep working.
This way our users will notice that the `apiKey` doesn't work and will have to switch to `loginWIthToken`.

Reply:
Oh ok, but we should remove the option from balena-request and balena-pine as well and just force users to move to using headers